### ISSUE TITLE
System.Collections/IVersionable.cls

### ISSUE THREAD

**Issue by MarkJohnstoneGitHub** [kellyethridge/VBCorLib#14]
*I'm submitting a bug report *
    Operating System: Windows [10]

System.Collections/IVersionable.cls 

Attribute VB_Name = "IVersionableCollection"

Shouldn't it be IVersionable?

**Comment by kellyethridge** [comment]
It's funny that I have both classes in the library. I clearly was just throwing stuff in several years ago. I'll have to address that. 

And yes, `IVersionable` seems like the more appropriate interface.

**Comment by MarkJohnstoneGitHub** [comment]
I'm checking that the impressive VBCorLib and trying to get it working under VBA 7. 
I picked that up when importing the classes.  It's going to take "some" time to wade through it all. 

Overall impressive! 🥇 

Wondering why didn't use predeclared classes?  If used predeclared classes you wouldn't need the Statics module and break encapsulation.  

As a result, could also move the factory methods in the Constructor class to the appropriate classes.  That way if only using sections of the VBCorLib, only the dependencies of the classes required would only have to be included in a custom application, instead of the whole VBCorLib package. 

For VBA 7 with the GetHashCode function possible requiring to return LongLong and conversions required on the appropriate classes to CLngLng or maybe even in VBA 6 conversion might also require using CLng.  I need to fully understand the GetHashCode function.

Using VBA 7 I was getting a TypeMismatch compile error for : 
GetHashCode = ObjPtr(CUnk(Obj))
That probably works under VB6 as returns a Long type as address references are treated as Longs.

Is the CorLib Type Library reference Addin Mac compatible?  

PS Rubberduck is impressed as well. :)

**Comment by kellyethridge** [comment]
Sounds like you have a project ahead of you with importing the classes. I have no experience with VBA passed Access97, and haven't touched it for at least 15 years. I only mess round in VB6 now when working on this library. It's sort of like my ship-in-a-bottle hobby.

I never built the library with the intent of the classes to be discrete. I was simply attempting to replicate many aspects of the .Net mscorlib library as a fun challenge, so everything depends on everything sometimes.

As for predeclared classes, there is no access to the attribute in the VB6 IDE, and I wasn't interested in going around the IDE. I didn't want to confuse people that see how the library code behaves vs what they can do from within the IDE. Even it I applied it to some classes, such as File, I wouldn't be able to apply it for all "static" classes because I wouldn't want to merge "static" methods and instance methods into a single class like merging ArrayList and ArrayListStatic.

I have no idea about the Mac :)

Well now I've heard of Rubberduck. I get a parse error. It doesn't seem to like a property named "Empty" that expose in my tests through a type library. I might have to submit a bug report.

**Comment by MarkJohnstoneGitHub** [comment]
Case of throw another project on the barbie. :)  
At this rate will never get to my original main project, thou that was getting unmanageable without some sort of framework.  I can see many components in the dotnet cor library that I've already worked on.  

I've been working slowly towards a framework of sorts a while back, which is half baked.  Was just about to revisit it and was looking to rehash ideas for an appropriate ICollection and ILIst interfaces.  The aim was to make it easier to enumerate and access items in repositories of lists containing different implementations ie. dictionaries/hashtables, arrays etc. 
 
Numerous times I've check out dotnet libraries for interface class ideas and having a consistent VB framework would make developing applications a lot easier, especially if ever decide to transition projects to dotnet.  Thankgod, I stumbled across this repository. 

VB6 and VBA basically the same except you can't create executables in VBA.  And a new type in VBA 7 like the LongPtr which is converted dependant on the version to Long or LongLong for handling addresses.   So I'll need to mindful when Longs are used that refer to an address handle. 

It is a royal pain exporting and importing classes to add and fix up attributes. Rubberduck has annotations to make like easier and it will prompt to fix up the attributes and keep them consistent.  You'll find many features that make life easier when developing in visual basic. From annotations for folder and attributes etc which will make it easier to organize such an extensive project.  And unit testing, code explorer etc.  Any suggestions for features and bug reports they will appreciate.  

In short, every VB developer needs a Rubberduck. :)
